_summaries/vaswani2017attention_1.md

```diff
@@ -1,19 +1,88 @@
 ---
 layout: summary
 title: Summary
-paper: {{paper_tag}}
-# Please fill out info below
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+paper: vaswani2017attention
+author: # samatharhay
+score: # 8
 ---
 
-TODO: Summarize the paper:
-* What is the core idea?
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+The paper introduced an architecture Transformers, an architecture solely based on attention.
+This architecture is meant to challenge more dominant models that rely on sequence aligned RNNs or convolutions.
+Unlike the aforenoted models the Transformer allows for significantly more parallelization, meaning faster training, while still producing state of the art results in translation tasks.
+
+## Architecture
+
+The Transformer uses stacked self-attention and point-wise fully connected layers for both the encoder and decoder.
+
+![Transformer](vaswani2017attention_1a.png)
+
+
+
+The Encoder is made up of a stack (N=6) of identical layers, each made up of two sub-layers.
+The first sub-layer is a multi-head self attention mechanism, and the second is a position wise fully connected FFN.
+A residual connection is employed around each sub-layer and followed by layer normalization.
+To facilitate the residual connections all sub-layers and embedding layers produce a fixed output (d_{model}=512).
+
+The Decoder is composed of a stack (N=6) of identical layers, made up of three sub-layers, two of which are the same as the encoder.
+The third sublayer, inserted before the self-attention layer, performs multi-head attention over the output of the encoder stack.
+All sub-layers have residual connections and normalization, like the sub-layers in the encoder.
+The self-attention layer has added masking to prevent positions from attending to subsequent positions, which helps ensure that predictions for position i can depend only on known outputs for positions before it.
+
+
+The Transformer uses two different types of attention: Scaled Dot-Product Attention and Multi-Head Attention.
+
+![Attention](vaswani2017attention_1b.png)
+
+Scaled Dot-Product Attention
+* Takes as input queries and keys of dimension dk and values of dimension dv.
```

Comment:
Please consider making this a bit clearer, maybe like so: "Takes queries (Q) and keys (K) of dimension $$d_k$$, and values (V) of dimension $$d_v$$ as inputs."